controllers/k8ssandra/medusa_reconciler.go

```diff
@@ -98,19 +98,22 @@ func (r *K8ssandraClusterReconciler) reconcileMedusa(
 
 		kcKey := utils.GetKey(kc)
 		// Create a cron job to purge Medusa backups
-		operatorNamespace := r.getOperatorNamespace()
-		purgeCronJob, err := medusa.PurgeCronJob(dcConfig, kc.SanitizedName(), operatorNamespace, logger)
-		if err != nil {
-			logger.Info("Failed to create Medusa purge backups cronjob", "error", err)
-			return result.Error(err)
-		}
-		purgeCronJob.SetLabels(labels.CleanedUpByLabels(kcKey))
-		recRes := reconciliation.ReconcileObject(ctx, remoteClient, r.DefaultDelay, *purgeCronJob)
-		switch {
-		case recRes.IsError():
-			return recRes
-		case recRes.IsRequeue():
-			return recRes
+		logger.Info("Checking if Medusa backups should be purged, PurgeBackups: " + fmt.Sprintf("%t", *medusaSpec.PurgeBackups))
+		if *medusaSpec.PurgeBackups {
```

Comment:
**Todo:** There's cleanup missing in case a cronjob has been created and backup purge is then turned off.
This means in case PurgeBackups is set to false explicitly, you need to look for the CronJob and delete it if it exists.